Add support for CentOS 7, add improved structure

stieg opened the pull request:
Addes support for CentOS 7 since they apparently don't like the
standard netcat name (jerks).  Instead they use nmap-ncat.  So
fix that and add better structure for parameters as per the puppet
way.

Also cleans up variable names a bit so meaning is clearer and more
standardized.

Have a look and tell me what you think.  First module I have contributed
back to so I don't know all the ways yet.  Any help is appreciated.

Commit:
Add support for CentOS 7, add improved structure

Addes support for CentOS 7 since they apparently don't like the
standard netcat name (jerks).  Instead they use nmap-ncat.  So
fix that and add better structure for parameters as per the puppet
way.

Also cleanus up variable names a bit so meaning is clearer and more
standardized.

rfletcher commented:
Thanks for the contribution. It looks good to me, except that I prefer not to change the `ensure` param to `package_ensure` at this point. I have seen `package_ensure` used elsewhere, but (I think) only when a class is managing multiple resources, as in the common package/config/service combination. This class is about as simple as it gets, and I'm not sure the change adds anything.

I probably could go either way there, but one more thing pushed me toward leaving the name alone: it's a breaking change, so I'd technically have to bump the version to 1.0.0 per Puppet guidelines.

Care to resubmit with the old name (or argue for keeping it)?

stieg commented:
I did it for consistency.  In this case since we don't have a need for the more formal names I'll just make them simpler.  Standby.

Commit:
Make the parameter names simpler

For backwards compatibility and because we don't need the fancy
namespaced names at this point in time in this project.

Commit:
Fix up comment too

stieg commented:
There we go.  Much better.
